app/controllers/events_controller.rb

```diff
@@ -0,0 +1,28 @@
+class EventsController < ApplicationController
+  include EventsHelper
+  before_action :authorization
+  helper_method :events_user_attending, :test
+
+  def index
+    @event = Event.all
+    @past_events = Event.past.paginate(page: params[:page], per_page: 2).order(created_at: :desc)
+    @upcoming_events = Event.upcoming.paginate(page: params[:page], per_page: 2).order(created_at: :desc)
```

Comment:
Please move the `paginate` and `order` preprocessing to the scope methods instead of writing them here.